src/test/java/com/solvd/AbstractTest.java

```diff
@@ -0,0 +1,47 @@
+package com.solvd;
+
+import com.solvd.utils.ConfigReader;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.chrome.ChromeOptions;
+import org.openqa.selenium.firefox.FirefoxOptions;
+import org.openqa.selenium.remote.RemoteWebDriver;
+import org.openqa.selenium.safari.SafariOptions;
+import org.testng.annotations.*;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.time.Duration;
+
+public class AbstractTest {
+    protected ThreadLocal<WebDriver> driverThreadLocal = new ThreadLocal<>();
```

Comment:
Why are you using driver with thread local? Is it required because you set driver in @beforeClass method?

Reply:
this is just a solution I found that allows me to run tests in parallel on multiple browsers